assets/js/components/dashboard-sharing/DashboardSharingDialog/Footer/index.js

```diff
@@ -0,0 +1,178 @@
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useCallback, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_MODULES } from '../../../../googlesitekit/modules/datastore/constants';
+import { CORE_UI } from '../../../../googlesitekit/datastore/ui/constants';
+import {
+	EDITING_USER_ROLE_SELECT_SLUG_KEY,
+	RESET_SETTINGS_DIALOG,
+	SETTINGS_DIALOG,
+} from '../../DashboardSharingSettings/constants';
+import useViewContext from '../../../../hooks/useViewContext';
+import { trackEvent } from '../../../../util';
+import Link from '../../../Link';
+import Button from '../../../Button';
+import Notice from './Notice';
+import Spinner from '../../../Spinner';
+import ErrorText from '../../../ErrorText';
+
+const { useSelect, useDispatch } = Data;
+
+export default function Footer( { closeDialog, openResetDialog } ) {
+	const viewContext = useViewContext();
+
+	const [ errorNotice, setErrorNotice ] = useState( null );
+	const [ isResetting, setIsResetting ] = useState( false );
+
+	const canSubmitSharingChanges = useSelect( ( select ) =>
+		select( CORE_MODULES ).canSubmitSharingChanges()
+	);
+	const isSaving = useSelect( ( select ) =>
+		select( CORE_MODULES ).isDoingSubmitSharingChanges()
+	);
+	const haveSharingSettingsChangedManagement = useSelect( ( select ) =>
+		select( CORE_MODULES ).haveSharingSettingsExpanded( 'management' )
+	);
+	const haveSharingSettingsChangedRoles = useSelect( ( select ) =>
+		select( CORE_MODULES ).haveSharingSettingsExpanded( 'sharedRoles' )
+	);
+	const settingsDialogOpen = useSelect(
+		( select ) => !! select( CORE_UI ).getValue( SETTINGS_DIALOG )
+	);
+	const resetDialogOpen = useSelect(
+		( select ) => !! select( CORE_UI ).getValue( RESET_SETTINGS_DIALOG )
+	);
+
+	const { saveSharingSettings } = useDispatch( CORE_MODULES );
+	const { setValue } = useDispatch( CORE_UI );
+	const { resetSharingSettings } = useDispatch( CORE_MODULES );
+
+	const onApply = useCallback( async () => {
+		setErrorNotice( null );
+
+		const { error } = await saveSharingSettings();
+		if ( error ) {
+			setErrorNotice( error.message );
+			return;
+		}
+
+		trackEvent( `${ viewContext }_sharing`, 'settings_confirm' );
+
+		// Reset the state to enable modules in when not editing or saving.
+		setValue( EDITING_USER_ROLE_SELECT_SLUG_KEY, undefined );
+
+		closeDialog();
+	}, [ viewContext, saveSharingSettings, setValue, closeDialog ] );
+
+	const onReset = useCallback( async () => {
+		setErrorNotice( null );
+		setIsResetting( true );
+
+		const { error } = await resetSharingSettings();
+		if ( error ) {
+			setErrorNotice( error.message );
+			return;
+		}
+
+		setIsResetting( false );
+
+		closeDialog();
+	}, [ closeDialog, resetSharingSettings ] );
+
+	const onCancel = useCallback( () => {
+		trackEvent( `${ viewContext }_sharing`, 'settings_cancel' );
+		closeDialog();
+	}, [ closeDialog, viewContext ] );
```

Comment:
Rather than calling the variables `onApply`, `onCancel`, etc., it might be nice to name them based on the action they perform. `onCancel` is really `closeResetDialog`, etc.

Reply:
Understood, thank you!